doc/source/rllib/rl-modules.rst

```diff
@@ -601,13 +644,30 @@ To find out, what APIs your Algorithms require, do the following:
 
 
 .. note::
-    You don't need the preceding VPG example module to implement any APIs because
-    you haven't considered training it with any particular algorithm.
-    You can find examples of algorithm-ready :py:class:`~ray.rllib.algorithms.ppo.PPO` custom RLModules
-    in the `tiny_atari_cnn_rlm example <https://github.com/ray-project/ray/blob/master/rllib/examples/rl_modules/classes/tiny_atari_cnn_rlm.py>`__
+
+    You didn't implement any APIs in the preceding example module, because
+    you hadn't considered training it with any particular algorithm yet.
+    You can find examples of custom :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` classes
+    implementing the :py:class:`~ray.rllib.core.rl_module.apis.self_supervised_loss_api.SelfSupervisedLossAPI` and thus
+    ready to train with :py:class:`~ray.rllib.algorithms.ppo.PPO` in the
+    `tiny_atari_cnn_rlm example <https://github.com/ray-project/ray/blob/master/rllib/examples/rl_modules/classes/tiny_atari_cnn_rlm.py>`__
     and in the `lstm_containing_rlm example <https://github.com/ray-project/ray/blob/master/rllib/examples/rl_modules/classes/lstm_containing_rlm.py>`__.
 
 
+You can mix supervised losses into any RLlib algorithm through the :py:class:`~ray.rllib.core.rl_module.apis.self_supervised_loss_api.SelfSupervisedLossAPI`.
+Your Learner actors automatically call the implemented
+:py:meth:`~ray.rllib.core.rl_module.apis.self_supervised_loss_api.SelfSupervisedLossAPI.compute_self_supervised_loss` method to compute the model's own loss
+passing it the outputs of the :py:meth:`~ray.rllib.core.rl_module.rl_module.RLModule.forward_train` call.
+
+
+See here for an `example script utilizing a self-supervised loss RLModule <https://github.com/ray-project/ray/blob/master/rllib/examples/curiosity/intrinsic_curiosity_model_based_curiosity.py>`__.
```

Comment:
I looked at this example on Saturday. This does a second `forward_train` in a connector. Has this changed?

Reply:
Still the same, but it's ok. It's not an additional forward pass, but *the* (only) forward pass for the ICM model.

Background: The ICM model receives no environment sample data under its own ModuleID, so any normal Learner would not even update it. The special learners we use in this example adds a special connector performing the forward pass through the ICM to get its predictions, then duplicates the batch for the "default_policy" to give the ICM something to learn (predict next state, etc..).